src/openms/source/FORMAT/FileHandler.cpp

```diff
@@ -834,6 +834,8 @@ if (first_line.hasSubstring("File	First Scan	Last Scan	Num of Scans	Charge	Monoi
       exp.getSourceFiles().push_back(src_file);
     }
 
+    exp.updateRanges();
```

Comment:
I think this is a good point and I could add it... I just also want to discuss if there is a better way (e.g., is it really a problem if it would recalculate e.g., MSLevels).
Would make a lot of things much safer as e.g. filtering or other processing may invalidate these in a non-transparent way.

Reply:
IMHO it is a bit of a problematic design to allow basically any container like change but rely on these "states".